scripts/generator/generator/pyclblast.py

```diff
@@ -0,0 +1,113 @@
+
+
+def generate_pyx(routine):
+    result = ""
+    if routine.implemented and routine.plain_name() and routine.level in ["1", "2a", "2b", "3"]:
+
+        result += SEPARATOR + NL
+        result += "# " + routine.description + ": " + routine.short_names() + NL
+        result += SEPARATOR + NL
+        result += NL
+
+        result += "cdef extern from \"clblast_c.h\":" + NL
+        np_dtypes = []
+        for flavour in routine.flavours:
+            if flavour.precision_name in ["S", "D", "C", "Z"]:
+                result += "    CLBlastStatusCode CLBlast" + flavour.name + routine.plain_name() + "("
+                result += ", ".join(routine.arguments_def_c(flavour)) + ","
+                result += "cl_command_queue* queue, cl_event* event)" + NL
+                np_dtypes.append(to_np_dtype(flavour))
+        result += "" + NL
+
+        buffers = routine.inputs[:] + routine.outputs[:]
+        result += "def " + routine.plain_name() + "(queue, "
+        result += ", ".join(routine.arguments_python()) + "):" + NL
+        result += "    dtype = check_dtype([" + ", ".join(buffers) + "], "
+        result += "[" + ", ".join(['"%s"' % d for d in np_dtypes]) + "])" + NL
+        for buf in buffers:
+            if buf in routine.buffers_vector():
+                result += "    check_vector("
+            else:
+                result += "    check_matrix("
+            result += buf + ", \"" + buf + "\")" + NL
+        result += "" + NL
+
+        for buf in buffers:
+            result += "    cdef cl_mem " + buf + "_buffer = <cl_mem><size_t>" + buf + ".base_data.int_ptr" + NL
+        result += "" + NL
+
+        result += "    cdef cl_command_queue command_queue = <cl_command_queue><size_t>queue.int_ptr" + NL
+        result += "    cdef cl_event event = NULL" + NL
+
+        for option in routine.options:
+            if option == "a_transpose":
+                result += "    a_transpose = CLBlastTransposeYes if a_transp else CLBlastTransposeNo" + NL
+            if option == "b_transpose":
+                result += "    b_transpose = CLBlastTransposeYes if b_transp else CLBlastTransposeNo" + NL
+            if option == "ab_transpose":
+                result += "    ab_transpose = CLBlastTransposeYes if ab_transp else CLBlastTransposeNo" + NL
+            if option == "side":
+                result += "    side = CLBlastSideRight if right_side else CLBlastSideLeft" + NL
+            if option == "triangle":
+                result += "    triangle = CLBlastTriangleLower if lower_triangle else CLBlastTriangleUpper" + NL
+            if option == "diagonal":
+                result += "    diagonal = CLBlastDiagonalUnit if unit_diagonal else CLBlastDiagonalNonUnit" + NL
+
+        result += "" + NL
+        result += "    cdef CLBlastStatusCode err" + NL
+        if_prefix = ""
+        for flavour in routine.flavours:
+            if flavour.precision_name in ["S", "D", "C", "Z"]:
```

Comment:
If you define the dict I was suggesting above as a global in the file then you wouldn't have to have these vars here it would be a compare to see if key in dict.